core/DataTable/Filter/Truncate.php

```diff
@@ -96,7 +96,7 @@ private function addSummaryRow($table)
             return;
         }
 
-        $rows   = $table->getRows();
+        $rows   = array_values($table->getRows());
```

Comment:
Truncate was buggy before if some rows were deleted and was summarizing wrong values under circumstances since it always reused the summary row when an array index was missing